CIPs/cip-145.md

```diff
@@ -0,0 +1,99 @@
+
+
+## Abstract
+<!--A short (~200 word) description of the technical issue being addressed.-->
+Abstract goes here.
+
+
+## Motivation
+<!--Motivation is critical for CIPs that want to change the Ceramic protocol. It should clearly explain why the existing protocol specification is inadequate to address the problem that the CIP solves. CIP submissions without sufficient motivation may be rejected outright.-->
+A Ceramic stream can fork if two different event both have the same CID in their `priv` field. 
+
+![lattice](../assets/cip-145/lattice.png)
+Fig. 1: A lattice where 1, 2, 3 are events that got anchored and A, B are events on a pruned branch. 2 and A both have `prev: 1's CID` creating a branch. 3 has a multi-prev `priv: [2's CID, B's CID]` joining the pruned branch back into the stream. 
+
+When a stream forks Ceramic uses a set of rules to determine witch of the multiple uncovered nodes will serve as the tip of the stream. The other uncovered events are on dead branches. If a pruned branch contains only time events and no data events it can be safely ignored. If a branch pruned branch contains data events then that data would be lost. In order to fix this we enable a event to specify multiple previous event so that a single event can cover more then one event.
+
+With a multiple previous event we can reduce the number of uncovered events and call any stream where there is only a single uncovered converged state. Any stream where there are data events without a common descendent are in a diverged state. The streams converged/diverged state can be determined just by looking at the `prev` field of the event headers of the events for that stream.
+
+* A join `A ⋁ B` find the common ancestor of the events `A` and `B`
+* A meet `A ⋀ B` find the common descendent of the events `A` and `B`
+
+## Specification
+<!--The technical specification should describe the syntax and semantics of any new feature.-->
+Specification goes here.
+
+
+## Rationale
+<!--The rationale fleshes out the specification by describing what motivated the design and why particular design decisions were made. It should describe alternate designs that were considered and related work, e.g. how the feature is supported in other languages. The rationale may also provide evidence of consensus within the community, and should discuss important objections or concerns raised during discussion.-->
+Rationale goes here.
+
+
+## Backwards Compatibility
+<!--All CIPs that introduce backwards incompatibilities must include a section describing these incompatibilities and their severity. The CIP must explain how the author proposes to deal with these incompatibilities. CIP submissions without a sufficient backwards compatibility section may be rejected outright.-->
+All events that have a string in `prev` instead of a list will work with this CIP.
+Nodes that don't have this CIP will not understand what new events with a `prev` list means.
```

Comment:
So it's a soft fork? New nodes will be able to understand old nodes, but not the other way around.

Reply:
Yes, new nodes should work with everything that already exists.

Please note that new data even on existing streams might be aggregated differently on newer nodes compared to older nodes.